src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/GerritTrigger.java

```diff
@@ -147,6 +147,8 @@
     private transient RunningJobs runningJobs = new RunningJobs();
     private List<GerritProject> gerritProjects;
     private List<GerritProject> dynamicGerritProjects;
+    //! Lock to control `dynamicGerritProjects`
+    private static final Object DYNAMIC_GERRIT_PROJECTS_LOCK = new Object();
```

Comment:
Are you sure `private final Object DYNAMIC_GERRIT_PROJECTS_LOCK = new Object();` didn't work?
This concerns me as now all Gerrit jobs on the master will need to wait for a single lock.

Reply:
Yes; you get a NullPointerException in `isInteresting` every time.

Reply:
Maybe a static Map with job names for keys and `Lock` objects as values?

Reply:
That would work once the jobs are in the map; are we guaranteed anything about one-thread-per-job or anything like that?

Reply:
Okay, as long as we never set `dynamicGerritProjects` to null once it's already set to something, then we're okay.  I've removed all the lock stuff.

Note that if we ever do set it to null after it was already a list, then we'll run into NPE problems.